Purchases/Public/Purchases.swift

```diff
@@ -993,15 +993,11 @@ public extension Purchases {
      */
     @available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
     @objc(purchasePackage:withDiscount:completion:)
-    func purchase(package: Package, discount: SKPaymentDiscount, completion: @escaping PurchaseCompletedBlock) {
-        // todo: add support for SK2 with discounts, move to new class
-        // https://github.com/RevenueCat/purchases-ios/issues/1055
-        guard let sk1Product = package.storeProduct.sk1Product else {
-            return
```

Comment:
This was not doing anything! 😅

Reply:
I'm confused, do you mean that the `guard let` wasn't doing anything or that the code wasn't doing anything if the package wasn't for a SK1 product? 
We weren't doing anything for SK2 products because we don't have support for promotional offers for SK2 yet

Reply:
But I was surprised we weren't at least logging something.